cmd/kogito/command/deploy/deploy_service.go

```diff
@@ -278,3 +319,149 @@ func (i *deployCommand) Exec(cmd *cobra.Command, args []string) (err error) {
 
 	return nil
 }
+
+// matchKogitoResources will verify if the given parameter is a expect drl or dmn file.
+// Accepted values are, supported file extensions as it is, directory or url
+// See supportedExtensions env
+// return the file content as io.Reader if the provided file or url is valid.
+func matchKogitoResources(resource string) (io.Reader, *buildv1.BinaryBuildRequestOptions, error) {
+	log := context.GetDefaultLogger()
+	options := &buildv1.BinaryBuildRequestOptions{}
+	var fileName string
+	// from url, do only basic check, for example, if the url has the suffix that's match the allowed ones
+	// if found download it and save on /tmp
+	switch {
+	case strings.HasPrefix(resource, "http"):
+		parsedURL, err := url.ParseRequestURI(resource)
+		if err != nil {
+			return nil, nil, fmt.Errorf("kogito resource is not a valid URL, received %s", resource)
+		}
+
+		hasExtension := false
+		ff := strings.Split(parsedURL.Path, "/")
+		fileName = strings.Join(strings.Fields(ff[len(ff)-1]), "")
+
+		if isSuffixSupported(fileName) {
+			hasExtension = true
+		}
+
+		if hasExtension {
+			response, err := http.Get(resource)
+			if err != nil {
+				return nil, nil, fmt.Errorf("failed to download %s, error message: %s", resource, err.Error())
+			}
+			options.AsFile = fileName
+			return response.Body, options, nil
+		}
+
+	// handle single file, files on dir.
+	default:
+		fileInfo, err := os.Stat(resource)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		if fileInfo.Mode().IsRegular() {
+			if isSuffixSupported(resource) {
+				log.Infof("File found %s.", resource)
+				ff := strings.Split(resource, "/")
+				fileName = strings.Join(strings.Fields(ff[len(ff)-1]), "")
+				fileReader, err := os.Open(resource)
+				if err != nil {
+					return nil, nil, err
+				}
+				options.AsFile = fileName
+				return fileReader, options, nil
+			}
+		}
+
+		if fileInfo.Mode().IsDir() {
```

Comment:
shouldn't be `else if` here ?
If `IsRegular()` is true and `isSuffixSupported(resource)`, we will still check if it is a dir ...

Reply:
make sense.